### PR TITLE
(PUP-7850) add task plan to language (for branch: feature_puppet-tasks)

### PR DESCRIPTION
**Pull request by hlindberg**

### CODE REVIEW

File: lib/puppet/pops/loader/puppet_plan_instantiator.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    # parse and validate
+    result = parser.parse_string(pp_code_string, source_ref)
+    # Only one function is allowed (and no other definitions)
+    case result.definitions.size
+    when 0
+      raise ArgumentError, _("The code loaded from %{source_ref} does not define the plan '%{plan_name}' - it is empty.") % { source_ref: source_ref, plan_name: typed_name.name }
+    when 1
+      # ok
+    else
+      raise ArgumentError, _("The code loaded from %{source_ref} must contain only the plan '%{type_name}' - it has additional definitions.") % { source_ref: source_ref, type_name: typed_name.name }
```
Comment (thallgren) [inline lib/puppet/pops/loader/puppet_plan_instantiator.rb]: nitpic, but please use %{plan_name} and not %{type_name} in all messages

File: lib/puppet/pops/model/model_tree_dumper.rb [hunk 2]
```diff
@@ -224,6 +224,17 @@ def dump_HostClassDefinition o
     result
   end
 
+  def dump_PlanDefinition o
```
Comment (thallgren) [inline lib/puppet/pops/model/model_tree_dumper.rb]: Need an entry for plan in `model/model_label_provider.rb` as well.

File: lib/puppet/pops/model/model_label_provider.rb [hunk 3]
```diff
@@ -60,6 +60,7 @@ def label_ConcatenatedString o          ; "Double Quoted String"              en
   def label_HeredocExpression o           ; "'@(#{o.syntax})' expression"       end
   def label_HostClassDefinition o         ; "Host Class Definition"             end
   def label_FunctionDefinition o          ; "Function Definition"               end
+  def label_PLanDefinition o              ; "Plan Definition"                   end
```
Comment (thallgren) [inline lib/puppet/pops/model/model_label_provider.rb]: this typo means that the method isn't found (L is uppercase)

Comment (hlindberg) [reply lib/puppet/pops/model/model_label_provider.rb]: meh

**Pull request merged**